lib/ecto/schema.ex

```diff
@@ -398,8 +401,9 @@ defmodule Ecto.Schema do
 
   """
   defmacro field(name, type \\ :string, opts \\ []) do
+    primary_key = Keyword.get(opts, :primary_key, false)
     quote do
-      Ecto.Schema.__field__(__MODULE__, unquote(name), unquote(type), false, unquote(opts))
```

Comment:
What if instead of passing `primary_key` as argument, we simply pass it as part of options and `__field__` can fetch it out? We just need to make sure to pass `primary_key: true` for the field in `@primary_key`.